climada_petals/hazard/landslide.py

```diff
@@ -37,43 +37,88 @@
+    """
+    Sample yearly events for a specified time span (n_years) from
+    a matrix representing annual occurrence probabilities per grid cell 
+    (prob_matrix). The matrix will usually be provided throught the
+    Landslide.from_prob() method and refer to grid-cells and their annual 
+    sliding probabilities.
+    Events are drawn from the specified distribution (binomial or 
+    poisson). 
+    Returns n_year events for the whole grid, each event being representative
+    of all slides happening in 1 year throughout the entire grid-area.
+    
     Parameters
     ----------
-    prob_matrix : scipy.sparse.csr matrix
-        matrix where each entry has an annual probability [0,1] of occurrence of
-        an event
+    prob_matrix : np.array()
+        matrix where each entry has a probability [0,1] of occurrence of
+        an event. Shape of array can be any and is determined by hazard area 
+        and resolution of probability-source-file when called through 
+        Landslide.from_prob().
     n_years : int
-        the timespan of the probabilistic simulation in years
-    dist : str
-        distribution to sample from. currently 'binom' (default) and 'poisson'
-
+        the timespan of the probabilistic simulation in years. Each year will
+        result in 1 event.
+    dist : str, 'binom' or 'poisson'
+        distribution to sample from. Default is 'binom'.
+    
     Returns
     -------
-    ev_matrix : scipy.sparse.csr matrix
-        csr matrix with number of success (events) from sampling process per
-        grid cell
-
+    scipy.sparse.csr_matrix : 
+        shape is (shape(prob_matrix),n_years). Each non-zero entry is 1 (re-
+        ferring to a 'hit' in that year and grid cell.) 
+    
     See also
     --------
-    set_ls_prob(), scipy.stats.binom.rvs(), scipy.stats.poisson.rvs()
-
+    sample_event_from_probs()
+    Landslide.from_prob()
     """
-    LOGGER.info('Sampling landslide events for a %i year period', n_years)
 
-    ev_matrix = prob_matrix.copy()
+    events = [
+        sample_event_from_probs(prob_matrix, n_years=1, dist=dist)
+        for i in range(n_years)
+        ]
+    return sparse.csr_matrix(events)
+    
+    
+def sample_event_from_probs(prob_matrix, n_years, dist):
+    """sample for a specified representative time span  (n_years) from
+    a matrix with occurrence probabilities. Draws from a specified distribution.
+
+    Parameters
+    ----------
+    prob_matrix : np.array()
+        matrix where each entry has a probability [0,1] of occurrence of
+        an event. Shape of array can be any and is determined by hazard area 
+        and resolution of probability-source-file when called through 
+        Landslide.from_prob().
+    n_years : int
+        the timespan of the probabilistic simulation in years. Will be 1 if
+        called from sample_events().
+    dist : str, 'binom' or 'poisson'
+        distribution to sample from.
```

Comment:
Please describe the parameters of the distributions. Binomial or Poisson is not enough.

Reply:
the formula of poisson / binomial distribution sampling is standard knowledge, i'd assume, and can be found explicitly written out in the description to the scipy implementation which is linked below. I added that the only shape params required are n_samples and prob_matrix for clarification (which can be seen in the formula below though, too)

Reply:
I think there might be a misunderstanding. There is no need to describe the functions, but the choice of parameters must be explained. There is no standard for that, and here they are defined under the hood.

Reply:
what parameters do you mean, then? the only ones requried are proability of success (given by prob_matrix) and number of samples to draw from (given by n_samples); all are user-given and explicit